plugins/experimental/traffic_dump/traffic_dump.cc

```diff
@@ -556,14 +710,21 @@ TSPluginInit(int argc, const char *argv[])
   info.support_email = "dev@trafficserver.apache.org";
 
   /// Commandline options
-  static const struct option longopts[] = {{"logdir", required_argument, nullptr, 'l'},
+  static const struct option longopts[] = {{"dump_body", optional_argument, nullptr, 'b'},
+                                           {"logdir", required_argument, nullptr, 'l'},
                                            {"sample", required_argument, nullptr, 's'},
                                            {"limit", required_argument, nullptr, 'm'},
+                                           {"client_ipv4", optional_argument, nullptr, '4'},
+                                           {"client_ipv6", optional_argument, nullptr, '6'},
                                            {nullptr, no_argument, nullptr, 0}};
   int opt                               = 0;
   while (opt >= 0) {
-    opt = getopt_long(argc, const_cast<char *const *>(argv), "l:", longopts, nullptr);
+    opt = getopt_long(argc, const_cast<char *const *>(argv), "l:s:m:4:6:b", longopts, nullptr);
```

Comment:
Do you need the short opts explicitly, if they're also embedded in the longopts?

Reply:
The short options string is needed, actually, even though on surface it seems redundant. By passing in `val` in the longopts structure ('b', 'l', etc.), a call to getopt_long for the long values could return the same value as the short, which is convenient for handling the long options the same way the short is handled in the switch statement. However, that does not define those short values as parsable short options on the command line. If you want short options, then they must be explicitly described in the short opts string.

Having said that, I'm glad you made me scrutinize this further. I realize I wasn't using required_argument, optional_argument, etc. correctly.